src/ring_theory/localization.lean

```diff
@@ -365,6 +387,99 @@ end
 lemma of.injective : function.injective (of : β → fraction_ring β) :=
 (is_add_group_hom.injective_iff _).mpr eq_zero_of
 
+section map
+open function is_ring_hom
+variables {A : Type u} [integral_domain A] [decidable_eq A]
+variables {B : Type v} [integral_domain B] [decidable_eq B]
+variables (f : A → B) [is_ring_hom f]
+
+def map (hf : injective f) : fraction_ring A → fraction_ring B :=
+quotient.lift (λ rs : A × (non_zero_divisors A), (f rs.1 : fraction_ring B) / (f rs.2.val)) $
+λ x y hxy,
+begin
+  erw div_eq_div_iff,
+  { rcases hxy with ⟨t, ht, H⟩,
+    replace H := ht _ H,
+    rw sub_eq_zero at H,
+    erw [← coe_mul, ← coe_mul, ← map_mul f, ← map_mul f],
+    congr' 2,
+    convert H.symm using 1; exact mul_comm _ _, },
+    all_goals { intro h,
+      rw [← coe_zero, ← map_zero f] at h,
+      exact ne_zero_of_mem_non_zero_divisors (by simp) (injective_comp of.injective hf h), },
+end
+
+@[simp] lemma map_coe (hf : injective f) (a : A) : map f hf a = f a :=
+begin
+  delta map,
+  erw quotient.lift_beta,
+  simp [map_one f],
+end
+
+@[simp] lemma map_of (hf : injective f) (a : A) : map f hf (of a) = of (f a) :=
+begin
+  delta map,
+  erw quotient.lift_beta,
+  simpa [map_one f],
+end
+
+@[simp] lemma map_mk (hf : injective f) (x : A × (non_zero_divisors A)) :
+  map f hf ⟦x⟧ = (f x.1) / (f x.2.val) := rfl
+
+@[simp] lemma map_circ_of (hf : injective f) :
+  map f hf ∘ (of : A → fraction_ring A) = (of : B → fraction_ring B) ∘ f :=
+funext $ map_of f hf
+
+instance map.is_field_hom (hf : injective f) : is_field_hom (map f hf) :=
+{ map_one := by erw [map_of f hf 1, map_one f]; refl,
+  map_mul :=
+  begin
+    rintros ⟨x⟩ ⟨y⟩,
+    repeat {erw map_mk},
+    rw div_mul_div,
+    congr' 1; erw [map_mul f, coe_mul],
+  end,
+  map_add :=
+  begin
+    rintros ⟨x⟩ ⟨y⟩,
+    repeat {erw map_mk},
+    rw div_add_div,
+    { congr' 1,
+      { simp only [coe_mul, coe_add, map_mul f, map_add f],
+        unfold_coes,
+        ring, },
+      { erw [← coe_mul, ← map_mul f],
+        refl, } },
+    all_goals { intro h,
+      rw [← coe_zero, ← map_zero f] at h,
+      exact ne_zero_of_mem_non_zero_divisors (by simp) (injective_comp of.injective hf h), },
+  end }
+
+def equiv_of_equiv (h : A ≃r B) : fraction_ring A ≃r fraction_ring B :=
```

Comment:
This can be generalized to localizations, I guess.

Reply:
Yes, that's a good point. I'll generalise onc I have Lean access